motoman_mh_support/urdf/mh80_macro.xacro

```diff
@@ -0,0 +1,183 @@
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_t.stl"/>
+      </geometry>
+    </collision>
+  </link>
+  <!-- end of link list -->
+
+  <!-- joint list -->
+  <joint name="${prefix}joint_1_s" type="revolute">
+    <origin xyz="0 0 0.540" rpy="0 0 0"/>
+    <parent link="${prefix}base_link"/>
+    <child link="${prefix}link_s"/>
+    <axis xyz="0 0 1"/>
+    <limit lower="${radians(-180)}" upper="${radians(180)}" effort="4903.32" velocity="${radians(170)}" />
+  </joint>
+  <joint name="${prefix}joint_2_l" type="revolute">
+    <origin xyz="0.145 0 0" rpy="0 0 0"/>
+    <parent link="${prefix}link_s"/>
+    <child link="${prefix}link_l"/>
+    <axis xyz="0 1 0" />
+    <limit lower="${radians(-90)}" upper="${radians(135)}" effort="5883.99" velocity="${radians(140)}" />
+  </joint>
+  <joint name="${prefix}joint_3_u" type="revolute">
+    <origin xyz="0 0 0.870" rpy="0 0 0"/>
+    <parent link="${prefix}link_l"/>
+    <child link="${prefix}link_u"/>
+    <axis xyz="0 -1 0"/>
+    <limit lower="${radians(-170)}" upper="${radians(251)}" effort="2986.13" velocity="${radians(160)}" />
+  </joint>
+  <joint name="${prefix}joint_4_r" type="revolute">
+    <origin xyz="0 0 0.210" rpy="0 0 0"/>
+    <parent link="${prefix}link_u"/>
+    <child link="${prefix}link_r"/>
+    <axis xyz="-1 0 0"/>
+    <limit lower="${radians(-360)}" upper="${radians(360)}" effort="772.27" velocity="${radians(230)}" />
+  </joint>
+  <joint name="${prefix}joint_5_b" type="revolute">
+    <origin xyz="1.025 0 0" rpy="0 0 0"/>
+    <parent link="${prefix}link_r"/>
+    <child link="${prefix}link_b"/>
+    <axis xyz="0 -1 0" />
+    <limit lower="${radians(-125)}" upper="${radians(125)}" effort="582.52" velocity="230" />
+  </joint>
+  <joint name="${prefix}joint_6_t" type="revolute">
+    <origin xyz="0 0 0" rpy="0 0 0"/>
+    <parent link="${prefix}link_b"/>
+    <child link="${prefix}link_t"/>
+    <axis xyz="-1 0 0" />
+    <limit lower="${radians(-360)}" upper="${radians(360)}" effort="250.07" velocity="${radians(350)}" />
+  </joint>
+  <!-- end of joint list -->
+
+  <!-- ROS base_link to Robot Manufacturer World Coordinates transform -->
+  <link name="${prefix}base" />
+  <joint name="${prefix}base_link-base" type="fixed">
+    <origin xyz="0 0 0.540" rpy="0 0 0"/>
+    <parent link="${prefix}base_link"/>
+    <child link="${prefix}base"/>
+  </joint>
+
+  <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
+  <link name="${prefix}flange"/>
+  <joint name="${prefix}joint_6_t-flange" type="fixed">
+    <origin xyz="0.175 0 0" rpy="0 0 0"/>
+    <parent link="${prefix}link_t"/>
+    <child link="${prefix}flange"/>
+  </joint>
+
+  <!-- ROS-Industrial 'tool0' frame: all-zeros tool frame -->
+  <link name="${prefix}tool0"/>
+  <joint name="${prefix}flange-tool0" type="fixed">
+    <origin xyz="0 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
```

Comment:
@EricMarcil: I believe this results in the correct orientation of `tool0`. Could you confirm?

Reply:
Yes, it is correct